### DISCUSSION TITLE
Boot Mode 2 (User Secure Boot) on PolarFire SoC FPGA (Icicle Kit)

### DISCUSSION THREAD

**Discussion by utku1998** [polarfire-soc/polarfire-soc-documentation#266] [General]
Hello, 

I am currently working with the "PolarFire SoC FPGA Icicle Kit". I have configured/flashed the device with the newest "Reference Design". Then, I have created an example test code (Simple LED Blinking example) in order test it for different boot modes (Boot Mode = 0, Boot Mode = 1, Boot Mode = 2, Boot Mode = 3). I have managed to boot the device with the Boot Modes = 0/1 and run the blinking code using the SoftConsole IDE. However I could not manage to boot the device in Boot Mode = 2 (User Secure Boot). While I am trying to flash my blinking code using SoftConsole IDEs Boot Mode = 2 option (with LIMRelease, which should be suitable for the User Secure Boot), I am getting the Error: "Bitstream or data is corrupted or noisy. ERROR_CODE: 801F". [Here](https://github.com/polarfire-soc/polarfire-soc-documentation/files/10219169/debugLog.txt), you can also see the "debugLog.txt".

Unfortunately, there is not enough decomentation about how to boot device with the Boot Mode = 2, but only small explanations for its purpose. Could you please help me and provide me a guidance/documentation which shows how I can boot the the device in Boot Mode = 2 (User Secure Boot)? 

Best regards,

**Comment by TommyMurphyTM1234** [comment]
> Unfortunately, there is not enough decomentation about how to boot device with the Boot Mode = 2

The issue is explained in the SoftConsole documentation here:

* https://mi-v-ecosystem.github.io/SoftConsole-Documentation/SoftConsole-v2022.2-RISC-V/using_softconsole/mpfs.html#known-issues-limitations

> _mpfsBootmodeProgrammer does not initialize or reset the SMK (sNVM Master Key) and assumes that this has been previously configured appropriately. If this is not the case (for example, for a new device) then an attempt to program boot mode 2 from SoftConsole will result in messages similar to the following and the SMK will need to be configured/initialized from Libero by programming an appropriate design that causes the SMK to be (re)generated:_
> ...
> ```
>programmer 'S201QVSUE' : device 'target' : EXPORT ERROR_CODE[16] = 801f programmer
> ```
